src/hashcash.cr

```diff
@@ -4,37 +4,52 @@ require "base64"
 require "digest/sha1"
 
 module Hashcash
-  VERSION = "0.1.0"
+  VERSION       = "0.1.0"
+  STAMP_VERSION = 1
+
+  # Hashcash.generate("resource")
+  # => 1:20:201206222555:resource::pOWgc88+uDuefr/o:MTMxNzg2MA==
+  # OR can customise hashcash defaults
+  # Hashcash.generate("hello", version: 1, bits: 16, date: Time.utc, ext: "goodbyye")
+  # => 1:16:201206222403:hello:goodbyye:kfwaGRadlD3ddc9G:MTMxMzY5NQ==
+  def self.generate(
+    resource : String, *,
+    version = 1,
+    bits = 20,
+    date = Time.utc,
+    ext = ""
+  )
+    Hashcash::Stamp.new(resource, version, bits, date, ext).generate
+  end
 
-  class Stamp
-    getter resource
-    getter bits
-    getter date
-    getter version : Int32
-    getter stamp_string : String
+  # Hashcash.verify("1:20:201206222555:resource::pOWgc88+uDuefr/o:MTMxNzg2MA==")
+  # => true
+  # Hashcash.verify("invalid_string")
+  # => false
+  def self.verify(hashcash_stamp : String, resource : String) : Bool
+    # make resource arg optional
+
+    # verifies the string return boolean
+    Hashcash::Stamp.new(resource).verify_stamp(hashcash_stamp)
+  end
 
-    STAMP_VERSION = 1 # move??
+  class Stamp
+    getter version, bits, date, resource, ext, stamp_string
 
     def initialize(
       @resource : String,
       @version = STAMP_VERSION,
       @bits = 20,
       @date = Time.utc,
-      @ext = ""
+      @ext = "",
+      @stamp_string = ""
```

Comment:
This is a double of up what the stamp itself represents. If you provide a [`#to_s`](https://crystal-lang.org/api/0.35.1/Object.html#to_s(io:IO):Nil-instance-method) implementation, this can then form the string representation.

Reply:
ext is "Extension (optional; ignored in version 1)." 
e.g. here:1:20:1303030600:anni@cypherspace.org::McMybZIhxKXu57jd:ckvi ext would just be the empty string between two colons.
Might not be necessary to store @string_stamp though. What do you think?

Reply:
Yep, there's no requirement to parse what's there (can just be captured as a string), but more details on the general usage intent here: http://www.hashcash.org/docs/hashcash.html#stamp-format__version_1_.